src/ORM/Association/BelongsToMany.php

```diff
@@ -1002,12 +1002,12 @@ protected function _junctionTableName($name = null)
     {
         if ($name === null) {
             if (empty($this->_junctionTableName)) {
-                $aliases = array_map('\Cake\Utility\Inflector::underscore', [
-                    $this->source()->alias(),
-                    $this->target()->alias()
+                $tablesNames = array_map('\Cake\Utility\Inflector::underscore', [
+                    $this->source()->table(),
+                    $this->target()->table()
                 ]);
-                sort($aliases);
-                $this->_junctionTableName = implode('_', $aliases);
+                sort($tablesNames);
+                $this->_junctionTableName = implode('_', $tablesNames);
```

Comment:
In the test I updated, authors has many articles, and articles belongsToMany tags, the belongsToMany association being defined in https://github.com/cakephp/cakephp/blob/master/tests/test_app/TestApp/Model/Table/ArticlesTable.php#L27